### PR TITLE
ENH: Extend RawImage wrapper to other data types

### PR DESCRIPTION
**Pull request by ferdymercury**
Fixes https://github.com/InsightSoftwareConsortium/ITK/issues/3184

## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by ferdymercury** [comment]
I copied the same structure than in https://github.com/InsightSoftwareConsortium/ITK/blob/047fa6934ac32711d1d7b84fa6eb81c72db01960/Modules/Filtering/ImageGrid/wrapping/itkResampleImageFilter.wrap#L7

File: Modules/IO/RAW/wrapping/itkRawImageIO.wrap [hunk 1]
```diff
@@ -1,5 +1,8 @@
 itk_wrap_class("itk::RawImageIO" POINTER)
+  UNIQUE(to_types "UC;${WRAP_ITK_SCALAR}")
   foreach(d ${ITK_WRAP_IMAGE_DIMS})
-    itk_wrap_template("${ITKM_F}${d}" "${ITKT_F},${d}")
+    foreach(t ${to_types})
+      itk_wrap_template("${ITKM_VI${t}${d}}${ITKM_VI${t}${d}}" "${ITKT_VI${t}${d}},${ITKT_VI${t}${d}}")
```
Comment (dzenanz) [inline Modules/IO/RAW/wrapping/itkRawImageIO.wrap:5]: `VI` means VectorImage, and this is a rarely used type.
```suggestion
      itk_wrap_template("${ITKM_${t}${d}}${ITKM_${t}${d}}" "${ITKT_${t}${d}},${ITKT_${t}${d}}")
```

**Review by dzenanz — CHANGES_REQUESTED** [review]

File: Modules/IO/RAW/wrapping/itkRawImageIO.wrap (hunk 1 above)
Comment (dzenanz) [reply Modules/IO/RAW/wrapping/itkRawImageIO.wrap:5]: Alternatively, use `itk_wrap_image_filter("${WRAP_ITK_SCALAR}" 2)` which will have about the same effect.

**Review by dzenanz — COMMENTED** [review]
Your commit message needs `ENH: ` prefix.

**Comment by ferdymercury** [comment]
> Your commit message needs `ENH: ` prefix.

Changed.

**Review by dzenanz — APPROVED** [review]

**Comment by ferdymercury** [comment]
There seems to be some errors:

```
2022-02-11T02:03:19.4372668Z /home/vsts/work/1/s-build/Wrapping/itkRawImageIO.cxx:26:30: error: expected expression
2022-02-11T02:03:19.4373891Z     typedef itk::RawImageIO< , > itkRawImageIO;
2022-02-11T02:03:19.4374578Z                              ^
2022-02-11T02:03:19.4375718Z /home/vsts/work/1/s-build/Wrapping/itkRawImageIO.cxx:27:30: error: expected expression
2022-02-11T02:03:19.4376797Z     typedef itk::RawImageIO< , >::Pointer itkRawImageIO_Pointer;
2022-02-11T02:03:19.4377483Z                              ^
2022-02-11T02:03:19.4378823Z /home/vsts/work/1/s-build/Wrapping/itkRawImageIO.cxx:29:30: error: expected expression
2022-02-11T02:03:19.4380060Z     typedef itk::RawImageIO< , > itkRawImageIO;
2022-02-11T02:03:19.4380756Z                              ^
2022-02-11T02:03:19.4381752Z /home/vsts/work/1/s-build/Wrapping/itkRawImageIO.cxx:30:30: error: expected expression
2022-02-11T02:03:19.4382647Z     typedef itk::RawImageIO< , >::Pointer itkRawImageIO_Pointer;
2022-02-11T02:03:19.4383601Z                              ^
2022-02-11T02:03:19.4384667Z /home/vsts/work/1/s-build/Wrapping/itkRawImageIO.cxx:32:30: error: expected expression
2022-02-11T02:03:19.4385754Z     typedef itk::RawImageIO< , > itkRawImageIO;
2022-02-11T02:03:19.4386719Z                              ^
2022-02-11T02:03:19.4387745Z /home/vsts/work/1/s-build/Wrapping/itkRawImageIO.cxx:33:30: error: expected expression
2022-02-11T02:03:19.4389324Z     typedef itk::RawImageIO< , >::Pointer itkRawImageIO_Pointer;
2022-02-11T02:03:19.4389896Z                              ^
2022-02-11T02:03:19.4390688Z /home/vsts/work/1/s-build/Wrapping/itkRawImageIO.cxx:35:30: error: expected expression
2022-02-11T02:03:19.4391407Z     typedef itk::RawImageIO< , > itkRawImageIO;
2022-02-11T02:03:19.4391882Z                              ^
2022-02-11T02:03:19.4392642Z /home/vsts/work/1/s-build/Wrapping/itkRawImageIO.cxx:36:30: error: expected expression
2022-02-11T02:03:19.4393502Z     typedef itk::RawImageIO< , >::Pointer itkRawImageIO_Pointer;
2022-02-11T02:03:19.4394029Z                              ^
2022-02-11T02:03:19.4395111Z /home/vsts/work/1/s-build/Wrapping/itkRawImageIO.cxx:38:30: error: expected expression
2022-02-11T02:03:19.4395848Z     typedef itk::RawImageIO< , > itkRawImageIO;
2022-02-11T02:03:19.4396450Z                              ^
2022-02-11T02:03:19.4397219Z /home/vsts/work/1/s-build/Wrapping/itkRawImageIO.cxx:39:30: error: expected expression
2022-02-11T02:03:19.4397928Z     typedef itk::RawImageIO< , >::Pointer itkRawImageIO_Pointer;
2022-02-11T02:03:19.4398443Z                              ^
2022-02-11T02:03:19.4399202Z /home/vsts/work/1/s-build/Wrapping/itkRawImageIO.cxx:41:30: error: expected expression
2022-02-11T02:03:19.4399931Z     typedef itk::RawImageIO< , > itkRawImageIO;
2022-02-11T02:03:19.4400405Z                              ^
2022-02-11T02:03:19.4401163Z /home/vsts/work/1/s-build/Wrapping/itkRawImageIO.cxx:42:30: error: expected expression
2022-02-11T02:03:19.4401887Z     typedef itk::RawImageIO< , >::Pointer itkRawImageIO_Pointer;
2022-02-11T02:03:19.4402390Z                              ^
2022-02-11T02:03:19.4402803Z 12 errors generated.
```

File: Modules/IO/RAW/wrapping/itkRawImageIO.wrap [hunk 2]
```diff
@@ -1,5 +1,8 @@
 itk_wrap_class("itk::RawImageIO" POINTER)
+  UNIQUE(to_types "UC;${WRAP_ITK_SCALAR}")
   foreach(d ${ITK_WRAP_IMAGE_DIMS})
-    itk_wrap_template("${ITKM_F}${d}" "${ITKT_F},${d}")
+    foreach(t ${to_types})
+      itk_wrap_template("${ITKM_${t}${d}}${ITKM_${t}${d}}" "${ITKT_${t}${d}},${ITKT_${t}${d}}")
+    endforeach()
   endforeach()
```
Comment (dzenanz) [inline Modules/IO/RAW/wrapping/itkRawImageIO.wrap:7]:
```suggestion
  itk_wrap_image_filter("${to_types}" 2)
```

Comment (dzenanz) [reply Modules/IO/RAW/wrapping/itkRawImageIO.wrap:7]: There seems to be something wrong with C++ template type specification. Let's try this shorter form.

**Review by dzenanz — APPROVED** [review]

**Review by thewtex — CHANGES_REQUESTED** [review]
@ferdymercury thanks for the contribution.

Please try the approach suggested inline.

The wrapping macros are explained in the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf). These should be used considering the declaration of RawImageIO:

`to_types` -> `types`

File: Modules/IO/RAW/wrapping/itkRawImageIO.wrap [hunk 3]
```diff
@@ -1,5 +1,4 @@
 itk_wrap_class("itk::RawImageIO" POINTER)
-  foreach(d ${ITK_WRAP_IMAGE_DIMS})
-    itk_wrap_template("${ITKM_F}${d}" "${ITKT_F},${d}")
-  endforeach()
+  UNIQUE(to_types "UC;${WRAP_ITK_SCALAR}")
+  itk_wrap_image_filter("${to_types}" 2)
```
[the remark this answers is not in the document]
Comment (ferdymercury) [reply Modules/IO/RAW/wrapping/itkRawImageIO.wrap]: thanks for the hint

**Review by thewtex — APPROVED** [review]
Thanks @ferdymercury

**Pull request merged**